pandas/core/frame.py

```diff
@@ -5818,13 +5818,12 @@ def _gotitem(self,
     def aggregate(self, func, axis=0, *args, **kwargs):
         axis = self._get_axis_number(axis)
 
-        # TODO: flipped axis
         result = None
-        if axis == 0:
-            try:
-                result, how = self._aggregate(func, axis=0, *args, **kwargs)
-            except TypeError:
-                pass
+        df = self if not axis else self.T
```

Comment:
Same thing?

Reply:
Yeah, a bit. ``df.agg`` and ``df.apply`` with ``axis=1`` give in many cases identical output, but not in all, as ``df.apply`` doesn't make the lookup in ``_cython_table``.

AFAIK, transposition is cheap in numpy/pandas, so this is an ok approach?

Reply:
I would rather you not do this, this is a quite hacky way of handling this, there is a small bug on the lower level i think.

Reply:
I don't think so, as ``df._aggregate`` currently doesn't take an ``axis`` parameter. I could add an ``axis`` parameter to ``df._aggregate`` and do the transposition there instead?

Reply:
you can try. this should be handled on a much lower level.

Reply:
I've added the ``axis`` parameter to ``df._aggregate``, so this is handled there.